### PR TITLE
Fix up PHPStan errors.

### PR DESCRIPTION
**Pull request by michbeck**
Road to level 8

### CODE REVIEW

File: src/Propel/Generator/Model/Column.php [hunk 1]
```diff
@@ -1739,6 +1739,7 @@ public static function generatePhpSingularName(string $phpName): string
     public function isContent(string $content): bool
     {
         $contentAttribute = $this->getAttribute('content');
+        // @todo: expression below has same operands - fix or kick?
 
         return $contentAttribute && strtoupper($contentAttribute) === strtoupper($contentAttribute);
```
Comment (dereuromark) [inline src/Propel/Generator/Model/Column.php:1744]: definitly doesnt make sense to have this nonsense operation/comparison :)

File: src/Propel/Runtime/ActiveQuery/SqlBuilder/AbstractSqlQueryBuilder.php [hunk 2]
```diff
@@ -127,7 +127,7 @@ public function buildParams(array $columnNames, ?Criteria $values = null): array
      * Build sql statement from a criteria and add it to the given statement collector.
      *
      * @param \Propel\Runtime\ActiveQuery\Criterion\AbstractCriterion $criterion
-     * @param array<mixed>|null $params
+     * @param array|null $params
```
Comment (dereuromark) [inline src/Propel/Runtime/ActiveQuery/SqlBuilder/AbstractSqlQueryBuilder.php:130]: Thats a regression IMO
We should keep the more specific annotation where we have them to signal the other places where we didnt check yet
The stan tools are wrong here to assume that those are equal, they are not in semantic meaning.

**Comment by dereuromark** [comment]
The other changes look good to me.

File: src/Propel/Generator/Model/Column.php (hunk 1 above)
Comment (mringler) [inline src/Propel/Generator/Model/Column.php:1744]: Oh, it has to be:

```suggestion
        return $contentAttribute && strtoupper($contentAttribute) === strtoupper($content);
```

Good find!

Comment (michbeck) [reply src/Propel/Generator/Model/Column.php:1744]: That's for sure ... but not sure if this is just a typo and someone wanted to compare $content with $contentAttribute?

File: src/Propel/Runtime/ActiveQuery/SqlBuilder/SelectQuerySqlBuilder.php [hunk 3]
```diff
@@ -105,7 +105,7 @@ protected function buildSelectClause(array &$sourceTableNamesCollector): string
     }
 
     /**
-     * @param array<mixed>|null $params
+     * @param array|null $params
```
Comment (dereuromark) [inline src/Propel/Runtime/ActiveQuery/SqlBuilder/SelectQuerySqlBuilder.php:108]: Same here IMO

**Review by dereuromark — APPROVED** [review]
Apart from the mixed annotations, this looks great.
Really awesome improvements going forward! Thx :)

**Comment by michbeck** [comment]
@mringler @dereuromark can we move forward here?

**Review by mringler — APPROVED** [review]
Oh, sorry, thank you for the reminder!

**Comment by mringler** [comment]
@michbeck It looks like @dereuromark has left the project, and it looks like there is no successor at Spryker at the moment. I have contacted the owner, and he told me that the project will keep being maintained by Spryker, but I haven't heard back if/when there will be someone who actively maintains Propel2, i.e. merges this PRs.
For what it's worth, I am very sorry for the situation! I really appreciate the work you put into this in the previous PRs, and I would have really liked to see this merged, preferably in acceptable time.
I guess when/if someone takes over, contributors like you and I can decide if they want to come back.
Best wishes

**Comment by floriankraemer** [comment]
@mringler thank you for the reminder and your patience! ❤️ I really hope you'll keep contributing after this, I'm really sorry that you had to wait, I know from personal experience that it is frustrating when your contributions won't move forward. 😞

I'll take this over _temporarely_ until I figure out who will be our new internal owner. Your PR looks good to me! 💪 There is just one [comment ](https://github.com/propelorm/Propel2/pull/1955#discussion_r1089065151)from Mark, to which I agree, that needs to be fixed but we could do that as well.

@michbeck can you maybe take care of it tomorrow? I'll approve the merge as soon as this has been fixed by someone.

File: src/Propel/Runtime/ActiveQuery/SqlBuilder/SelectQuerySqlBuilder.php (hunk 3 above)
Comment (PhilinTv) [inline src/Propel/Runtime/ActiveQuery/SqlBuilder/SelectQuerySqlBuilder.php:108]:
```suggestion
     * @param array<mixed>|null $params
```

**Comment by floriankraemer** [comment]
@gechetspr this is ready for merge now.

**Pull request merged**